pkg/api/types.go

```diff
@@ -2154,3 +2154,32 @@ type RangeAllocation struct {
 	// a single allocated address (the fifth bit on CIDR 10.0.0.0/8 is 10.0.0.4).
 	Data []byte `json:"data"`
 }
+
+// LockSpec is the specification of a resource lock.
+type LockSpec struct {
```

Comment:
If it's used only by the client, why does HeldBy need to be in the API at all? Just use annotations and resourceVersion-based optimistic concurrency.

In my experience, where there is a single blob of opaque data, people eventually want structured / key-value data instead: DNS TXT, "notes" and "description" strings, HeldBy (e.g., in Chubby, clients stuff various protobufs), ...